### PR TITLE
fix(FEC-11151): quality dropdown is cropped when user clicks on ‘Quality’ settings on the player after scrolling down

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

update the player client rect on document.scroll event

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/shell/shell.js [hunk 1]
```diff
@@ -219,12 +219,19 @@ class Shell extends Component {
    */
   componentDidMount() {
     const {player, eventManager} = this.props;
+    eventManager.listen(
+      document,
+      'scroll',
+      debounce(() => {
```
Comment (dan-ziv) [inline src/components/shell/shell.js:225]: remove the anonymous function and use arrow function for the callback `_onDocumentScroll`

File: src/components/shell/shell.js [hunk 2]
```diff
@@ -233,7 +240,7 @@ class Shell extends Component {
       FakeEvent.Type.RESIZE,
       debounce(() => {
```
Comment (dan-ziv) [inline src/components/shell/shell.js:241]: also here

File: src/components/shell/shell.js [hunk 3]
```diff
@@ -219,12 +219,19 @@ class Shell extends Component {
    */
   componentDidMount() {
```
Comment (dan-ziv) [inline src/components/shell/shell.js:220]: remove the listeners on `componentWillUnmount`

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: src/components/shell/shell.js (hunk 3 above)
Comment (yairans) [reply src/components/shell/shell.js:220]: automatically made by https://github.com/kaltura/playkit-js-ui/blob/master/src/event/with-event-manager.js#L20

File: src/components/shell/shell.js [hunk 4]
```diff
@@ -246,14 +253,33 @@ class Shell extends Component {
...
+    }
+  }
+
+  /**
+   * document scroll handler
+   *
+   * @returns {void}
+   * @memberof Shell
+   */
+  _onDocumentScroll(): void {
```
Comment (dan-ziv) [inline src/components/shell/shell.js:268]: If we don't have any more logic in it, no need this function, define just `_updatePlayerClientRect` as a callback for the document scroll event.

**Review by dan-ziv — APPROVED** [review]

**Comment by dan-ziv** [comment]
@yairans don't we need to change in `menu.js` the use of `guiClientRect` to `playerClientRect`?

**Comment by yairans** [comment]
No. Gui updated from player

**Review by dan-ziv — APPROVED** [review]

**Pull request merged**